src/main/java/frc/trigon/robot/subsystems/climber/ClimberVisualization.java

```diff
@@ -0,0 +1,163 @@
+        final double targetStringLength = calculateStringLengthMeters(targetPositionRotations);
+        final Rotation2d currentStringAngle = calculateStringAngle(currentFirstJointPitch.minus(ClimberConstants.FIRST_JOINT_ANGLE_ADDITION), currentStringLength);
+        final Rotation2d targetStringAngle = calculateStringAngle(targetFirstJointPitch.minus(ClimberConstants.FIRST_JOINT_ANGLE_ADDITION), targetStringLength);
+
+        setCurrentMechanism2dState(currentFirstJointPitch, currentStringAngle, currentStringLength);
+        setTargetMechanism2dState(targetFirstJointPitch, targetStringAngle, targetStringLength);
+        log(currentFirstJointPitch);
+    }
+
+    private void setCurrentMechanism2dState(Rotation2d currentFistJointPitch, Rotation2d currentStringAngle, double currentStringLength) {
+        currentFirstJointLigament.setAngle(flipAngle(currentFistJointPitch));
+        currentStringPositionLigament.setAngle(flipAngle(currentStringAngle));
+        currentStringPositionLigament.setLength(currentStringLength);
+
+    }
+
+    /**
+     * Sets the target state of the mechanism, but doesn't log the mechanism.
+     *
+     * @param targetFirstJointPitch the target pitch of the first joint
+     * @param targetStringAngle     the string angle of the string
+     * @param targetStringLength    the target string length of the climber
+     */
+    private void setTargetMechanism2dState(Rotation2d targetFirstJointPitch, Rotation2d targetStringAngle, double targetStringLength) {
+        targetFirstJointLigament.setAngle(flipAngle(targetFirstJointPitch));
+        targetStringPositionLigament.setAngle(flipAngle(targetStringAngle));
+        targetStringPositionLigament.setLength(targetStringLength);
+    }
+
+    /**
+     * Logs the mechanism, and the poses.
+     *
+     * @param currentPitch the current pitch or the climber
+     */
+    private void log(Rotation2d currentPitch) {
+        Logger.recordOutput(key, mechanism);
+        Logger.recordOutput("Poses/Components/" + name + "FirstJointPose", calculateClimberFirstJointPose(firstJointOriginPoint, currentPitch));
+        Logger.recordOutput("Poses/Components/" + name + "secondJointPose", calculateClimberSecondJointPose(calculateClimberFirstJointPose(firstJointOriginPoint, currentPitch), currentState));
+    }
+
+
+    private Pose3d calculateClimberSecondJointPose(Pose3d firstJointPose, ClimberConstants.ClimberState currentState) {
+        if (currentState.affectedByRobotWeight) {
+            Transform3d climberTransform = new Transform3d(
+                    new Translation3d(0, ClimberConstants.DISTANCE_BETWEEN_JOINTS_METERS, 0),
+                    new Rotation3d(0, 90, 0)
+            );
+            return firstJointPose.transformBy(climberTransform);
+        }
+        Transform3d climberTransform = new Transform3d(
+                new Translation3d(0, ClimberConstants.DISTANCE_BETWEEN_JOINTS_METERS, 0),
+                new Rotation3d(0, 0, 0)
+        );
```

Comment:
make the transforms final
no need for the whole thing in the if. The only thing that changes is the pitch (have a ternary operator for it). 
If we're affected by weight, the pitch should be `90 - the first joint pitch` (assuming it's degrees. keep in mind rotation3d returns radians)